Make export llama checkpoint and param optional

jackzhxng opened the pull request:
Summary: Refactor to make checkpoint path and param path optional.

Differential Revision: D71404805

larryliu0820 commented on examples/models/llama/model_args.py:
```diff
@@ -8,7 +8,7 @@ class ModelArgs:
     n_heads: int = 32
     n_kv_heads: Optional[int] = None
-    vocab_size: int = -1  # defined later by tokenizer
+    vocab_size: int = 512  # Arbitrary value, should be defined later by tokenizer.
```
Why do we set it to arbitrary value?

jackzhxng replied:
We need an actual value to fall back on when a params file isn't provided so that the embedding layer can be instantiated

larryliu0820 commented on examples/models/llama/export_llama_lib.py:
```diff
@@ -191,7 +184,6 @@ def build_args_parser() -> argparse.ArgumentParser:
     parser.add_argument(
         "-c",
         "--checkpoint",
-        default=f"{ckpt_dir}/params/demo_rand_params.pth",
```
add `required = False`?

larryliu0820 commented on examples/models/llama/export_llama_lib.py:
```diff
@@ -273,7 +265,6 @@ def build_args_parser() -> argparse.ArgumentParser:
     parser.add_argument(
         "-p",
         "--params",
-        default=f"{ckpt_dir}/params/demo_config.json",
```
Same here.

larryliu0820 commented on examples/models/llama/export_llama_lib.py:
```diff
@@ -273,7 +265,6 @@ def build_args_parser() -> argparse.ArgumentParser:
         "-p",
         "--params",
-        default=f"{ckpt_dir}/params/demo_config.json",
         help="config.json",
```
Talk about the behavior when this option is not provided

jackzhxng committed:
Make export llama checkpoint and param optional (#9456)

Summary:
Pull Request resolved: https://github.com/pytorch/executorch/pull/9456

Refactor to make checkpoint path and param path optional.

Reviewed By: larryliu0820

Differential Revision: D71404805
